pow/src/main/java/tech/pegasys/teku/pow/fallback/strategy/PriorityEth1ProviderSelector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow.fallback.strategy;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import tech.pegasys.teku.pow.Eth1Provider;
+
+public class PriorityEth1ProviderSelector implements Eth1ProviderSelector {
+
+  final List<Eth1Provider> candidates;
+  private final AtomicReference<Eth1Provider> bestCandidate;
+  private final AtomicInteger bestCandidateIndex;
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public PriorityEth1ProviderSelector(final List<Eth1Provider> candidates) {
+    assert candidates != null && !candidates.isEmpty();
```

Comment:
nit: Should use `Predicates.checkArgument` rather than the `assert` keyword here (`assert` keyword is skipped by default which gives a false sense of security).

Reply:
Done